### ISSUE TITLE
browser does not redirect to login page after setting password as new user 

### ISSUE THREAD

**Issue by individual-it** [owncloud/core#32647]
### Steps to reproduce
1. make sure owncloud is installed in the root folder of the webserver e.g. `http://localhost/`
2. create a new user with username and password
3. receive the set-password link for the new user
4. open this link an set a password

### Expected behaviour
redirection to login page

### Actual behaviour
redirection to same page, and an error shown that the token in not valid
the password was actually set and the token deleted, that is all correct, the issue is the redirection

in https://github.com/owncloud/core/blob/stable10/settings/js/setpassword.js#L39 `OC.getRootPath()` is called, but if oC runs in the root of the webserver that function returns an empty string.

see also failing tests on drone https://drone.owncloud.com/owncloud/core/10258/444

### Server configuration
**Operating system**:
Ubuntu 18.04
**Web server:**
Apache2
**Database:**
sqlite
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
10.0.10
**Updated from an older ownCloud or fresh install:**
fresh
**Where did you install ownCloud from:**
git

**The content of config/config.php:**

```
{
    "system": {
        "license-key": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "172.17.0.1"
        ],
        "dbtype": "sqlite3",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "instanceid": "ocjsvlzs93hf",
        "htaccess.RewriteBase": "\/",
        "enable_certificate_management": true,
        "trusted_proxies": {
            "1": "127.0.0.1"
        },
        "datadirectory": "\/home\/artur\/www\/owncloud-core\/data",
        "overwrite.cli.url": "http:\/\/localhost\/",
        "version": "10.0.10.0",
        "installed": true,
        "theme": "",
        "loglevel": 0,
        "maintenance": false,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1025",
        "skeletondirectory": "\/home\/artur\/www\/owncloud-core\/tests\/acceptance\/..\/..\/apps\/testing\/data\/apiSkeleton"
    }
}

```

#### ownCloud log (data/owncloud.log)
```
{"reqId":"g2xnKByvZom2MRXruhS6","level":3,"time":"2018-09-10T06:49:34+00:00","remoteAddr":"172.18.0.1","user":"admin","app":"settings","method":"GET","url":"\/settings\/users\/setpassword\/form\/053574324765919158293\/ana","message":"Exception: {\"Exception\":\"OCP\\\\InvalidUserTokenException\",\"Message\":\"The token provided is invalid.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/settings\\\/Controller\\\/UsersController.php(523): OC\\\\Settings\\\\Controller\\\\UsersController->checkPasswordSetToken('053574324765919...', 'ana')\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(153): OC\\\\Settings\\\\Controller\\\\UsersController->setPasswordForm('053574324765919...', 'ana')\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(85): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Settings\\\\Controller\\\\UsersController), 'setPasswordForm')\\n#3 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(100): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Settings\\\\Controller\\\\UsersController), 'setPasswordForm')\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('UsersController', 'setPasswordForm', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(342): OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(909): OC\\\\Route\\\\Router->match('\\\/settings\\\/users...')\\n#7 \\\/var\\\/www\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#8 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/settings\\\/Controller\\\/UsersController.php\",\"Line\":564}"}
```

CC @sharidas @davitol

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/24697 (Can't login LDAP users. 'Wrong password' returned), https://github.com/owncloud/core/issues/7936 (Wrong redirect on login), https://github.com/owncloud/core/issues/28129 (Login screen: entering the wrong password breaks the redirect), https://github.com/owncloud/core/issues/16431 (Recover login password), and https://github.com/owncloud/core/issues/7356 (Redirect after login breaks when wrong password entered after session timeout).

**Comment by individual-it** [comment]
similar issue when using `.htaccess`
1. install server in subfolder e.g. `http://localhost/owncloud-core`
2. make sure `.htaccess` file is writable
3. run ` occ config:system:set htaccess.RewriteBase --value /owncloud-core`
4. run `maintenance:update:htaccess`
5. make sure `.htaccess` was changed

now the redirection happens to: https://localhost/owncloud-core
correct would be https://localhost/owncloud-core/ or https://localhost/owncloud-core/login

**Comment by individual-it** [comment]
fixed by #32648
